app/views/users/_activity_stats.html.erb

```diff
@@ -0,0 +1,30 @@
+<div class="row mu-tab-body">
+  <div class="col-md-6">
+    <%= render partial: 'activity_indicator',
+               locals: {
+                 title: t(:exercises),
+                 stats: [{name: t(:solved_exercises_count), value: @activity[:exercises][:solved_count]},
+                         {name: t(:solved_exercises_percentage), value: "#{@activity[:exercises][:solved_count] / @activity[:exercises][:count] * 100}%"}]
+               } %>
+
+    <%= render partial: 'activity_indicator',
+               locals: {
+                 title: t(:forum),
+                 stats: [{name: t(:messages), value: @activity[:messages][:count]},
+                         {name: t(:approved_messages), value: @activity[:messages][:approved]}]
+               } %>
+  </div>
+
+  <div class="col-md-3 col-md-offset-3">
+    <ul class="nav nav-pills nav-stacked">
+      <li <%= !params[:date_from] && 'class=active' %>>
```

Comment:
We usually try to avoid accessing params directly and set an instance variable in the controller to later access it in the view.
Also it would be nice to extract it to a helper!